### PR TITLE
report(redesign): order gauges - default, pwa, plugins

### PR DESCRIPTION
**Pull request by Hoten**
#8185

### CODE REVIEW

File: lighthouse-core/report/html/renderer/report-renderer.js [hunk 1]
```diff
@@ -223,20 +223,23 @@ class ReportRenderer {
...
       for (const category of report.reportCategories) {
         const renderer = specificCategoryRenderers[category.id] || categoryRenderer;
         const categoryGauge = renderer.renderScoreGauge(category, report.categoryGroups || {});
 
-        // Group gauges that aren't default at the end of the header
-        if (renderer.renderScoreGauge === categoryRenderer.renderScoreGauge) {
-          defaultGauges.push(categoryGauge);
+        if (Util.isPluginCategory(category.id)) {
+          pluginGauges.push(categoryGauge);
+        } else if (renderer.renderScoreGauge === categoryRenderer.renderScoreGauge) {
```
Comment (patrickhulce) [inline lighthouse-core/report/html/renderer/report-renderer.js]: this seems like an interesting way to tell, I guess the point is "do we render this gauge exactly like the others or does the gauge look different"?

Comment (Hoten) [reply lighthouse-core/report/html/renderer/report-renderer.js]: i 100% don't understand `renderer.renderScoreGauge === categoryRenderer.renderScoreGauge`.

File: lighthouse-core/report/html/renderer/report-renderer.js [hunk 2]
```diff
@@ -223,20 +223,23 @@ class ReportRenderer {
...
+      const customGauges = []; // PWA.
+      const pluginGauges = [];
       for (const category of report.reportCategories) {
         const renderer = specificCategoryRenderers[category.id] || categoryRenderer;
         const categoryGauge = renderer.renderScoreGauge(category, report.categoryGroups || {});
 
-        // Group gauges that aren't default at the end of the header
-        if (renderer.renderScoreGauge === categoryRenderer.renderScoreGauge) {
-          defaultGauges.push(categoryGauge);
+        if (Util.isPluginCategory(category.id)) {
```
Comment (exterkamp) [inline lighthouse-core/report/html/renderer/report-renderer.js]: Lol great `Util` function!

File: lighthouse-core/report/html/renderer/report-renderer.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/report/html/renderer/report-renderer.js]: we weren't alone 😆 https://github.com/GoogleChrome/lighthouse/pull/6526/files#r234373117

File: lighthouse-core/test/report/html/renderer/report-renderer-test.js [hunk 3]
```diff
@@ -104,21 +104,38 @@ describe('ReportRenderer', () => {
...
         return el.matches('.lh-gauge--pwa__wrapper');
       });
 
+      const indexOfPluginGauge = Array.from(output
+        .querySelectorAll('.lh-scores-header > a[class*="lh-gauge"]')).findIndex(el => {
+        return el.matches('.lh-gauge__wrapper--plugin');
+      });
+
+      assert.notEqual(-1, indexOfPwaGauge);
+      assert.notEqual(-1, indexOfPluginGauge);
```
Comment (patrickhulce) [inline lighthouse-core/test/report/html/renderer/report-renderer-test.js]: seems like we also want to assert that `indexOfPwaGauge` is less than `indexOfPluginGauge`? and/or that they're the last two gauges?

Comment (Hoten) [reply lighthouse-core/test/report/html/renderer/report-renderer-test.js]: done

File: lighthouse-core/report/html/renderer/report-renderer.js [hunk 4]
```diff
@@ -223,20 +223,23 @@ class ReportRenderer {
     // }
 
     if (scoreHeader) {
-      const defaultGauges = [];
-      const customGauges = [];
+      // Group gauges in this order: mainstream, pwa, plugins.
+      const mainstreamGauges = [];
```
Comment (brendankenny) [inline lighthouse-core/report/html/renderer/report-renderer.js]: can we still use "deault" (from `default-config.js`) or "core" for these?

File: lighthouse-core/report/html/renderer/report-renderer.js (hunk 1 above)
Comment (brendankenny) [reply lighthouse-core/report/html/renderer/report-renderer.js]: if it's confusing, should keep the comment :) (or make it more informative!)

File: lighthouse-core/report/html/renderer/report-renderer.js (hunk 4 above)
Comment (Hoten) [reply lighthouse-core/report/html/renderer/report-renderer.js]: done

File: lighthouse-core/report/html/renderer/report-renderer.js (hunk 1 above)
Comment (Hoten) [reply lighthouse-core/report/html/renderer/report-renderer.js]: done

**Pull request merged**